shell/platform/darwin/ios/framework/Source/FlutterPlatformPlugin.mm

```diff
@@ -153,6 +155,37 @@ - (void)setSystemChromePreferredOrientations:(NSArray*)orientations {
                   userInfo:@{@(kOrientationUpdateNotificationKey) : @(mask)}];
 }
 
+- (NSArray*)getSystemChromePreferredOrientations {
```

Comment:
> getPreferredOrientations returns a list of device orientations as they were requested in setPreferredOrientations

Does it actually do this?  It seems like it's ignoring whatever was previously set and is instead always returning the UIViewController supported orientations.

Reply:
Thanks for the review! I really appreciated code snippets, because i'm not confident in my obj-c skills 😅

Hmm, I'm making the assumption that requested orientations are stored in `_orientationPreferences` (inside `FlutterViewController`) and returned in `supportedInterfaceOrientations`. 

Now I'm thinking about adding a public field `flutterRequestedOrientation` in the `FlutterViewController` and using it in the`getSystemChromePreferredOrientations`

Reply:
Is there any point to putting this in the embedder then?  Can't that just be tracked in the dart code?

Reply:
Yes, now it looks like an over-complicated logic, compared to a simple solution that I filed to the framework repo - https://github.com/flutter/flutter/pull/123474. I feel awkward for wasting your time, although I really appreciate all of your comments

Reply:
No problem at all, thank you for thinking it all through!

Does this mean this PR should be closed in favor of https://github.com/flutter/flutter/pull/123474?

Reply:
Looks like https://github.com/flutter/flutter/pull/123474 was closed, so it sounds like this should be too for the same reasons?  You may want to start with a concrete design proposal that can be iterated on.